### PR TITLE
Implementation fixes and tests

### PR DESCRIPTION
**Pull request by roxblnfk**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️/❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

* [x] more tests
* [x] documentation

### CODE REVIEW

**Comment by roxblnfk** [comment]
[Fix injector to allow specifying named parameters in any order](https://trello.com/c/HITlInXp/52-fix-injector-to-allow-specifying-named-parameters-in-any-order)

File: src/InvalidParameterException.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Injector;
+
+use Psr\Container\ContainerExceptionInterface;
+
+class InvalidParameterException extends \InvalidArgumentException implements ContainerExceptionInterface
```
Comment (samdark) [inline src/InvalidParameterException.php]: Is it really related to Container?

File: src/MissingRequiredArgumentException.php [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+declare(strict_types=1);
```
Comment (samdark) [inline src/MissingRequiredArgumentException.php:3]: Do we need to handle InvalidParameterException  and MissingRequiredArgumentException separately?

Maybe it's enough to have \InvalidArgumentException only?

Comment (samdark) [reply src/MissingRequiredArgumentException.php:3]: Decided to leave as is for the possibility of introducing friendly exception.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**